CMakeLists.txt

```diff
@@ -204,7 +204,7 @@ include_directories(SYSTEM
   ${OPENGL_INCLUDE_DIR}
   ${PYTHON_INCLUDE_PATH}
 )
-include_directories(src ${catkin_INCLUDE_DIRS})
+include_directories(src SYSTEM ${catkin_INCLUDE_DIRS})
```

Comment:
Hmm, there is this:

```
#pragma GCC system_header
    This pragma takes no arguments. It causes the rest of the code in the current file to be treated as if it came from a system header. See System Headers. 
```
But the rest of the file is too broad. If that could be turned off again it would be perfect...